src/client/Client.cc

```diff
@@ -5799,6 +5817,7 @@ void Client::unmount()
 
   while (!mds_requests.empty()) {
     ldout(cct, 10) << "waiting on " << mds_requests.size() << " requests" << dendl;
+    flush_mdlog_sync();
```

Comment:
yeah, you are right, flush_mdlog_sync() in the while loop would be trigged N times 
I should call this before this while loop and add check in flush_mdlog_sync